Zend/zend_multiply.h

```diff
@@ -290,6 +290,19 @@ static zend_always_inline size_t zend_safe_address(size_t nmemb, size_t size, si
 	return (size_t) res;
 }
 
+#elif defined(_MSC_VER)
+
+static zend_always_inline size_t zend_safe_address(size_t nmemb, size_t size, size_t offset, bool *overflow)
+{
+	size_t res;
+	if (UNEXPECTED(FAILED(ULongLongMult(nmemb, size, &res)) || FAILED(ULongLongAdd(res, offset, &res)))) {
```

Comment:
LGTM for 64 bits, but a bit less sure it's appropriate for 32 bits with size_t though, ie UIntMult ?

Reply:
Ah yes saw it but I thought the api you mentioned above made a real difference, thanks for the pointer.

Reply:
See https://godbolt.org/z/K43dPhavE for the comparision; the intrinsic versions saves an `addc` at the cost of a couple more instructions.